languages/javalang/builder-renderer/java-builder-client/src/main/kotlin/io/vrap/codegen/languages/javalang/client/builder/requests/JavaHttpRequestRenderer.kt

```diff
@@ -191,15 +193,20 @@ class JavaHttpRequestRenderer @Inject constructor(override val vrapTypeProvider:
             |   httpRequestPath += "?" + String.join("&", params);
             |}
         """.trimMargin().escapeAll()
-        
+        val bodySetter: String = if(bodyName != null){
+            if(this.bodies[0].type.isFile())
+                "try{httpRequest.setBody(Files.readAllBytes(file.toPath()));}catch(Exception e){e.printStackTrace();}"
```

Comment:
And the lines:
```java
httpRequest.setBody(Files.readAllBytes(file.toPath()));
```
load the file into memory first, ideally we would stream the contents here. But may be it's fine for now 😉

Reply:
Yeah  using multipart upload might be the best way, but i just did it like this for sake of simplicity. we might come back to this.

Reply:
> Yeah using multipart upload might be the best way, but i just did it like this for sake of simplicity. we might come back to this if it causes trouble.

Reply:
Found out the reason, it was actually that we dont have a logger abstraction/convention. maybe i can leave it as here and create a separate issue for logging @katmatt ?